pageserver/src/tenant/timeline.rs

```diff
@@ -5010,6 +5036,10 @@ impl Timeline {
         // 1. it is older than cutoff LSN;
         // 2. it is older than PITR interval;
         // 3. it doesn't need to be retained for 'retain_lsns';
+
+        // TODO(yuchen): we could consider current retain_lsns as infinite leases.
```

Comment:
feel free to create a tracking issue if this is something you plan to work on in short term